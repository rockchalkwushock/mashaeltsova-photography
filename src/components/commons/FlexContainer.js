import styled from 'styled-components'

const FlexContainer = styled.div`
  align-items: ${({ align }) => align};
  display: flex;
  flex-direction: ${({ direction }) => direction};
  justify-content: ${({ justify }) => justify};
  text-align: ${({ tAlign }) => tAlign};
`

FlexContainer.defaultProps = {
  align: 'inherit',
  direction: 'row',
  justify: 'inherit',
  tAlign: 'inherit'
}
FlexContainer.displayName = 'FlexContainer'

export default FlexContainer
