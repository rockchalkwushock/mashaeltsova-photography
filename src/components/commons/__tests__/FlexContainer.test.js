import React from 'react'
import { mount, shallow } from 'enzyme'

import FlexContainer from '../FlexContainer'

describe('Styled Component: <FlexContainer />', () => {
  test('should render markup', () => {
    const wrapper = mount(<FlexContainer />)
    expect(wrapper).toMatchSnapshot()
  })
  test('should have default props', () => {
    const wrapper = mount(<FlexContainer />)
    expect(wrapper.props()).toHaveProperty('align', 'inherit')
    expect(wrapper.props()).toHaveProperty('direction', 'row')
    expect(wrapper.props()).toHaveProperty('justify', 'inherit')
    expect(wrapper.props()).toHaveProperty('tAlign', 'inherit')
  })
  test('should have default styles', () => {
    const wrapper = shallow(<FlexContainer />)
    expect(wrapper).toHaveStyleRule('align-items', 'inherit')
    expect(wrapper).toHaveStyleRule('flex-direction', 'row')
    expect(wrapper).toHaveStyleRule('justify-content', 'inherit')
    expect(wrapper).toHaveStyleRule('text-align', 'inherit')
  })
  test('should have updated styles', () => {
    const wrapper = shallow(
      <FlexContainer
        align="center"
        direction="column"
        justify="center"
        tAlign="center"
      />
    )
    expect(wrapper).toHaveStyleRule('align-items', 'center')
    expect(wrapper).toHaveStyleRule('flex-direction', 'column')
    expect(wrapper).toHaveStyleRule('justify-content', 'center')
    expect(wrapper).toHaveStyleRule('text-align', 'center')
  })
})
