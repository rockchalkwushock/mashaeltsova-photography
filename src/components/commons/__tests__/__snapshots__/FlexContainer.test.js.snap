// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Component: <FlexContainer /> should render markup 1`] = `
.c0 {
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: inherit;
  -webkit-justify-content: inherit;
  -ms-flex-pack: inherit;
  justify-content: inherit;
  text-align: inherit;
}

<FlexContainer
  align="inherit"
  direction="row"
  justify="inherit"
  tAlign="inherit"
>
  <div
    className="c0"
    direction="row"
  />
</FlexContainer>
`;
